src/lib/core/SymbolSolver.cpp

```diff
@@ -493,86 +512,99 @@ void SymbolSolver::solveNames(LinuxProcMapEntry * procMapEntry)
 			lst.push_back(&it->second);
 		}
 	}
+	if ( addr2lineCmd.str().length() > prefixLen) {
+	  //hide error if silent
+	  if (gblOptions != NULL && gblOptions->outputVerbosity <= MALT_VERBOSITY_DEFAULT)
+		    addr2lineCmd << ' ' << "2>/dev/null";
+	  theCommands.push_back(addr2lineCmd.str());
+	}
+
+	// Uncomment to see the command split
+	// for (auto &s : theCommands) {
+	//	   std::cerr << "__COMMAND_________________________________________" << std::endl;
+	//	   std::cerr << s << std::endl;
+	// }
 	
-	//debug
-	if (gblOptions != NULL && gblOptions->outputVerbosity >= MALT_VERBOSITY_VERBOSE)
-		printf("MALT: %s\n",addr2lineCmd.str().c_str());
-	
-	//hide error if silent
-	if (gblOptions != NULL && gblOptions->outputVerbosity <= MALT_VERBOSITY_DEFAULT)
-		 addr2lineCmd << ' ' << "2>/dev/null";
-	
+
+
+	//printf("MALT: %s\n",addr2lineCmd.str().c_str());
+
 	//if no extry, exit
 	if (!hasEntries)
 		return;
-	
+
 	//run command
 	//std::cerr << addr2lineCmd.str() << std::endl;
-	FILE * fp = popen(addr2lineCmd.str().c_str(),"r");
-	
-	//check error, skip resolution
-	if (fp == NULL)
-	{
-		MALT_ERROR_ARG("Fail to use addr2line on %1 to load symbols : %2.").arg(elfFile).argStrErrno().end();
-		return;
-	}
-	
-	//read all entries (need big for some big template based C++ application, 
-	//seen at cern)
-	static char bufferFunc[200*4096];
-	static char bufferFile[20*4096];
 	size_t i = 0;
-	while (!feof(fp))
-	{
-		//read the two lines
-		char * funcRes = fgets(bufferFunc,sizeof(bufferFunc),fp);
-		char * fileRes = fgets(bufferFile,sizeof(bufferFile),fp);
-		
-		if (funcRes != bufferFunc || fileRes != bufferFile)
-			break;
-
-		//std::cerr << bufferFunc;
-		//std::cerr << bufferFile;
-
-		//check end of line and remove it
-		int endLine = strlen(bufferFunc);
-		assumeArg(bufferFunc[endLine-1] == '\n',"Missing \\n at end of line for the function or symbol name read from addr2line : %1.").arg(bufferFunc).end();
-		bufferFunc[endLine-1] = '\0';
-
-		//check errors
-		assume(i < lst.size(),"Overpass lst size.");
-
-		//search ':' separator at end of "file:line" string
-		char * sep = strrchr(bufferFile,':');
-		if (sep == NULL)
-		{
+	for (auto &cmd : theCommands) {
+		//debug
+		if (gblOptions != NULL && gblOptions->outputVerbosity >= MALT_VERBOSITY_VERBOSE)
```

Comment:
Fine, I would just keep tab-based indent to keep consistency with the rest of the code.